src/RevProc.cc

```diff
@@ -1911,16 +1958,23 @@ bool RevProc::ClockTick( SST::Cycle_t currentCycle ){
   if(!Pipeline.empty() &&
      (Pipeline.front().second.cost > 0)){
     Pipeline.front().second.cost--;
-    if((Pipeline.front().second.cost == 0)){ // &&
+    if(Pipeline.front().second.cost == 0){ // &&
       // Ready to retire this instruction
-      uint16_t tID = Pipeline.front().first;
+      uint16_t HartID = Pipeline.front().first;
       output->verbose(CALL_INFO, 6, 0,
-                      "Core %d ; ThreadID %d; Retiring PC= 0x%" PRIx64 "\n",
-                      id, tID, ExecPC);
+                      "Core %" PRIu16 "; Hart %" PRIu16 "; Thread %" PRIu32 "; Retiring PC= 0x%" PRIx64 "\n",
+                      id, HartID, ThreadOnHart(HartID)->GetThreadID(), ExecPC);
       Retired++;
-      DependencyClear(tID, &(Pipeline.front().second));
+
+      // Only clear the dependency if there is no outstanding load
+      if((Pipeline.front().second.rd != 0) && 
+        (RegFile->GetLSQueue()->count(make_lsq_hash(Pipeline.front().second.rd,
+                                                                 InstTable[Pipeline.front().second.entry].rdClass,
+                                                                 HartID))) == 0){
```

Comment:
THIS?

Reply:
Yup. This was the fix for not clearing the scoreboard when there is a memreq in flight. Ugly as sin but with the way the LSQueue works I think this is the best way to do it. Open to ideas 🤷